### PR TITLE
Change to use SHALLOW_DUPLICATE_ATTRIB() since that's API since R3.3.0

### PR DESCRIPTION
**Pull request by MichaelChirico**
Follow-up to #6183, part of #6257. See also this comment thread: https://github.com/Rdatatable/data.table/commit/cd89cebd8d3df077c0849da4b964527046e6e83b#r144215033

### CODE REVIEW

File: src/dogroups.c [hunk 1]
```diff
@@ -482,8 +482,13 @@ SEXP keepattr(SEXP to, SEXP from)
   // Same as R_copyDFattr in src/main/attrib.c, but that seems not exposed in R's api
   // Only difference is that we reverse from and to in the prototype, for easier calling above
   SET_ATTRIB(to, ATTRIB(from));
```
Comment (ben-schwen) [inline src/dogroups.c:484]: Note that `SET_OBJECT` will also be deprecated. Maybe we should switch to `SHALLOW DUPLICATE ATTRIBUTE` here

void SHALLOW_DUPLICATE_ATTRIB(SEXP to, SEXP from) {
    SET_ATTRIB(CHK(to), shallow_duplicate(CHK(ATTRIB(CHK(from)))));
    SET_OBJECT(CHK(to), OBJECT(from));
    IS_S4_OBJECT(from) ?  SET_S4_OBJECT(to) : UNSET_S4_OBJECT(to);
}

File: src/assign.c [hunk 2]
```diff
@@ -153,7 +153,7 @@ static SEXP shallow(SEXP dt, SEXP cols, R_len_t n)
   SEXP newdt = PROTECT(allocVector(VECSXP, n)); protecti++;   // to do, use growVector here?
   SET_ATTRIB(newdt, shallow_duplicate(ATTRIB(dt)));
   SET_OBJECT(newdt, OBJECT(dt));
```
Comment (ben-schwen) [inline src/assign.c:155]: Backport `SHALLOW_DUPLICATE_ATTRIB` and use it here?

https://github.com/wch/r-source/blob/8e4314e5929e2a2f890fda48301324865b79d2c9/src/main/memory.c#L3944-L3948

Comment (MichaelChirico) [reply src/assign.c:155]: Nice find... I think we can just go ahead and use it?

image

I would leave this as a TODO for now, given the experience with `isDataFrame()`/ #6235 -- let's wait for the `NOTE` to start throwing at least.

Comment (MichaelChirico) [reply src/assign.c]: Per the comment just below, it's available from 3.3.0 (which we now use), so I went ahead and made the switch.

**Comment by MichaelChirico** [comment]
ping @ben-schwen  :)

**Comment by ben-schwen** [comment]
LGTM

**Pull request merged**